examples/rcn.py

```diff
@@ -0,0 +1,390 @@
+print(f"Creating factors took {end-start:.3f} seconds.")
+
+
+# %% [markdown]
+# # 5. Run inference
+
+# %% [markdown]
+# ## 5.1 Helper functions to initialize the evidence for a given image
+
+# %%
+def get_bu_msg(img: np.ndarray) -> np.ndarray:
+    """Computes the bottom-up messages given a test image.
+
+    Args:
+        img: The rgb image to compute bottom up messages on (H x W x 3).
+
+    Returns:
+        An array of shape [16 x H x W] denoting the presence or absence of an oriented line segment at a particular location.
+            The elements of this array belong to the set {+1, -1}.
+    """
+
+    num_orients = 16
+    brightness_diff_threshold = 40.0
+
+    filtered = np.zeros((filters.shape[0],) + img.shape, dtype=np.float32)
+    for i in range(filters.shape[0]):
+        kern = filters[i, :, :]
+        filtered[i] = fftconvolve(img, kern, mode="same")
+
+    localized = np.zeros_like(filtered)
+    cross_orient_max = filtered.max(0)
+    filtered[filtered < 0] = 0
+    for i, (layer, suppress_mask) in enumerate(zip(filtered, suppression_masks)):
+        competitor_maxs = maximum_filter(layer, footprint=suppress_mask, mode="nearest")
+        localized[i] = competitor_maxs <= layer
+    localized[cross_orient_max > filtered] = 0
+
+    # Threshold and binarize
+    localized *= (filtered / brightness_diff_threshold).clip(0, 1)
+    localized[localized < 1] = 0
+
+    pooled_channel_weights = [(0, 1), (-1, 1), (1, 1)]
+    pooled_channels = [-np.ones_like(sf) for sf in localized]
+    for i, pc in enumerate(pooled_channels):
+        for channel_offset, factor in pooled_channel_weights:
+            ch = (i + channel_offset) % num_orients
+            pos_chan = localized[ch]
+            if factor != 1:
+                pos_chan[pos_chan > 0] *= factor
+            np.maximum(pc, pos_chan, pc)
+
+    bu_msg = np.array(pooled_channels)
+    bu_msg[bu_msg == 0] = -1
+    return bu_msg
+
+
+# %% [markdown]
+# ## 5.1.1 Visualizing bu_msg for a sample image
+
+# %% [markdown]
+# bu_msg has shape (16, H, W) where each 1 <= f <= 16 denotes the present or absense of a oriented edge
+
+# %%
+r_test_img = test_set[4]
+r_bu_msg = get_bu_msg(r_test_img)
+img = np.ones((200, 200))
+fig, ax = plt.subplots(1, 2, figsize=(20, 10))
+ax[0].imshow(r_test_img, cmap="gray")
+ax[0].axis("off")
+ax[0].set_title("Input image", fontsize=30)
+for i in range(r_bu_msg.shape[0]):
+    img[r_bu_msg[i] > 0] = 0
+
+ax[1].imshow(img, cmap="gray")
+ax[1].axis("off")
+ax[1].set_title("Max filter response across 16 channels", fontsize=30)
+fig.tight_layout()
+
+
+# %% [markdown]
+# ## 5.2 Run map product inference on all test images
+
+# %%
+def get_evidence(bu_msg: np.ndarray, frc: np.ndarray):
+    """Function to get evidence
+
+    Args:
+        bu_msg: Array of shape (n_features, 200, 200). Contains BU messages
+        frc: Array of shape (n_frcs, 3).
+
+    Returns:
+        evidence; Array of shape (n_frcs, M). Contains evidence
+    """
```

Comment:
```suggestion
    """Returns the evidence (shape (n_frcs, M))

    Args:
        bu_msg: Array of shape (n_features, 200, 200). Contains BU messages
        frc: Array of shape (n_frcs, 3).
    """
```
You can make this concise if you want

> Returns: (or Yields: for generators)
>    Describe the type and semantics of the return value. If the function only returns None, this section is not required. It may also be omitted if the docstring starts with Returns or Yields (e.g. """Returns row from Bigtable as a tuple of strings.""") and the opening sentence is sufficient to describe the return value. 

https://google.github.io/styleguide/pyguide.html

Reply:
Changed to this style.